src/services/services.ts

```diff
@@ -5008,6 +5015,94 @@ namespace ts {
             return getDefinitionFromSymbol(type.symbol, node);
         }
 
+         /// Goto implementation
+        function getImplementationAtPosition(fileName: string, position: number): ImplementationLocation[] {
+            synchronizeHostData();
+
+            const node = getTouchingPropertyName(getValidSourceFile(fileName), position);
+            const typeChecker = program.getTypeChecker();
+
+            // If invoked directly on a shorthand property assignment, then return
+            // the declaration of the symbol being assigned (not the symbol being assigned to).
+            if (node.parent.kind === SyntaxKind.ShorthandPropertyAssignment) {
+                const result: ReferenceEntry[] = [];
+                getReferenceEntryForShorthandPropertyAssignment(node, typeChecker, result);
+                return result.length > 0 ? result : undefined;
+            }
+            else if (node.kind === SyntaxKind.SuperKeyword || isSuperProperty(node.parent)) {
+                // References to and accesses on the super keyword only have one possible implementation, so no
+                // need to "Find all References"
+                const symbol = typeChecker.getSymbolAtLocation(node);
+                if (symbol.valueDeclaration) {
+                    return [getReferenceEntryFromNode(symbol.valueDeclaration)];
+                }
+            }
+            else {
+                const entries: ImplementationLocation[] = [];
+                // Perform "Find all References" and retrieve only those that are implementations
+                const result = getReferencedSymbolsForNode(node, program.getSourceFiles(), /*findInStrings*/false, /*findInComments*/false, /*implementations*/true);
+
+                forEach(result, referencedSymbol => {

```

Comment:
This could just be
```ts
return flatMap(referencedSymbols, symbol =>
    symbol.references.map(({textSpan, fileName}) => ({textSpan, fileName})));
```